cmd/tle/commands/encrypt.go

```diff
@@ -42,164 +42,81 @@ func Encrypt(flags Flags, dst io.Writer, src io.Reader, network *http.Network) e
 		return tlock.Encrypt(dst, src, flags.Round)
 
 	case flags.Duration != "":
-		durations, err := parseDurations(flags.Duration)
+		start := time.Now()
+		totalDuration, err := parseDurationsAsSeconds(start, flags.Duration)
 		if err != nil {
 			return err
 		}
-		now := time.Now()
-		decryptionTime := durations.from(now)
-		if decryptionTime.Before(now) || decryptionTime.Equal(now) {
+
+		decryptionTime := start.Add(time.Duration(totalDuration) * time.Second)
+		if decryptionTime.Before(start) || decryptionTime.Equal(start) {
 			return ErrInvalidDurationValue
 		}
+
 		roundNumber := network.RoundNumber(decryptionTime)
 		return tlock.Encrypt(dst, src, roundNumber)
 	default:
 		return errors.New("you must provide either duration or a round flag to encrypt")
 	}
 }
 
-type combinedDuration struct {
-	seconds int
-	minutes int
-	hours   int
-	weeks   int
-	days    int
-	months  int
-	years   int
-}
-
 var ErrDuplicateDuration = errors.New("you cannot use the same duration unit specifier twice in one duration")
 
-func (c *combinedDuration) apply(value int, multiplier DurationMultiplier) error {
-	switch multiplier {
-	case Second:
-		if c.seconds != 0 {
-			return ErrDuplicateDuration
-		}
-		c.seconds = value
-	case Minute:
-		if c.minutes != 0 {
-			return ErrDuplicateDuration
-		}
-		c.minutes = value
-	case Hour:
-		if c.hours != 0 {
-			return ErrDuplicateDuration
-		}
-		c.hours = value
-	case Day:
-		if c.days != 0 {
-			return ErrDuplicateDuration
-		}
-		c.days = value
-	case Week:
-		if c.weeks != 0 {
-			return ErrDuplicateDuration
-		}
-		c.weeks = value
-	case Month:
-		if c.months != 0 {
-			return ErrDuplicateDuration
-		}
-		c.months = value
-	case Year:
-		if c.years != 0 {
-			return ErrDuplicateDuration
-		}
-		c.years = value
-	}
-	return nil
-}
-
-func (c *combinedDuration) from(someTime time.Time) time.Time {
-	return someTime.AddDate(
-		c.years, c.months, c.days+(c.weeks*7),
-	).Add(
-		time.Duration(c.minutes) * time.Minute,
-	).Add(
-		time.Duration(c.hours) * time.Hour,
-	).Add(
-		time.Duration(c.seconds) * time.Second,
-	)
-}
+func parseDurationsAsSeconds(start time.Time, input string) (float64, error) {
```

Comment:
I think this should be returning a duration directly, no?

```suggestion
func parseDurationsAsSeconds(start time.Time, input string) (time.Duration, error) {
```

Reply:
Ah nice catch - have updated it in all the places